source/dem/dem.cc

```diff
@@ -548,44 +499,73 @@ DEMSolver<dim>::load_balance()
 
 template <int dim>
 inline bool
-DEMSolver<dim>::no_load_balance()
+DEMSolver<dim>::is_load_balance_iteration()
 {
-  return false;
+  if (parameters.model_parameters.load_balance_method ==
```

Comment:
those if/else if are being check at every time step. 

+ or minus 5 if at every time step won't affect the code performance, so it is ok for this PR. 

I don't think it is clean. I already have a "clean" implementation thank to Paul. We could discuss this a your return. :)

Reply:
I like Olivier's new implementation. It is indeed way cleaner. But this can be merged as is as a checkpoint to his implementation.